### PR TITLE
add source file and source line number in html result.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @ansyral @qinezh @fenxu 
#450

### CODE REVIEW

**Comment by ansyral** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/HtmlRenderer.cs [hunk 1]
```diff
@@ -343,5 +357,18 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken
...
 
         #endregion
+
+        #region Protected Methods
+
+        protected static StringBuffer AppendSourceInfo(StringBuffer result, IMarkdownRenderer renderer, IMarkdownToken token)
+        {
+            if (renderer.Options.ShouldExportSourceInfo)
+            {
+                result = result + " sourceFile=\"" + StringHelper.HtmlEncode(token.SourceInfo.File) + "\" sourceLineNumber=\"" + token.SourceInfo.LineNumber.ToString() + "\"";
```
Comment (fenxu) [inline src/Microsoft.DocAsCode.MarkdownLite/HtmlRenderer.cs]: >result = result + " sourceFile=\"" + StringHelper.HtmlEncode(token.SourceInfo.File) + "\" sourceLineNumber=\"" + token.SourceInfo.LineNumber.ToString() + "\""; [](start = 16, length = 159)

Shall we split them for better perf?

**Comment by fenxu** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/HtmlRenderer.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.MarkdownLite/HtmlRenderer.cs]: (sb + s) -> sb
sb + s1 + s2 ... + sn -> sb + s2 ... + sn -> ... -> sb

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**